### PR TITLE
feat(20102): e2e testing payments landing page

### PR DESCRIPTION
**Pull request by nnorbert**
e2e testing payments landing page.

### CODE REVIEW

File: apps/payments/e2e/specs/landingPage/llandingPage.spec.ts
Comment (rgrnearform) [inline apps/payments/e2e/specs/landingPage/llandingPage.spec.ts:1]: is the name llandingPage.spec.ts correct?

File: apps/payments/e2e/objects/LandingPage.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  private readonly headerCTA: Locator;
+  private readonly footerCTA: Locator;
+  private readonly header: Locator;
+  private readonly feedbackLink: Locator;
+
+  private readonly title = "Payments";
+  private readonly headerBlockContent =
+    "Our Payments Building Block is built on a robust infrastructure that enables secure and efficient financial transactions, giving users peace of mind every time they use it.";
+
+  private readonly benefitsHeader = "Benefits";
```
Comment (rgrnearform) [inline apps/payments/e2e/objects/LandingPage.ts:14]: I would create an array of benefits and a loop to reduce repetitive code.

e.g: 

private readonly benefits = [
    {
      title: "Seamless Integration",
      content: "The Payments Building Block seamlessly connects with multiple payment methods, including credit cards, digital wallets, and bank transfers, to offer flexibility to your users.",
    },
    {
      title: "Enhanced Security",
      content: "The Payments Building Block employs encryption protocols and payment providers that are Payment Card Industry (PCI) compliant, giving you peace of mind.",
    },
    {
      title: "User-Friendly Interface",
      content: "The Payments Building Block has an intuitive interface that makes payment processing simple for both administrators and users. Whether you're generating payment requests or tracking transactions, it's quick and simple.",
    },
    {
      title: "Accessibility",
      content: "The Payments Building Block is designed with accessibility in mind. It conforms with the Web Content Accessibility Guidelines (WCAG) 2.0, ensuring that all users can easily access the service.",
    },
    {
      title: "Dedicated Support",
      content: "Our team of experts is here to assist you every step of the way. Whether you have technical questions or need assistance with implementation, you can rely on the Payments Building Block's dedicated support team.",
    }
  ];

async checkBenefits() {
    for (const { title, content } of this.benefits) {
      const heading = this.page.getByRole("heading", { name: title });
      const textContent = this.page.getByText(content);
      await expect(heading).toBeVisible();
      await expect(textContent).toBeVisible();
    }
  }

File: apps/payments/e2e/objects/LandingPage.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  async goto(url?: string) {
+    const gotoUrl = url ?? landingPageUrl;
+    await this.page.goto(gotoUrl);
+  }
+
+  async checkHeader() {
+    await expect(this.header).toBeVisible();
+  }
+
+  async checkHeaderCTA(link: string) {
```
Comment (rgrnearform) [inline apps/payments/e2e/objects/LandingPage.ts:63]: Generic CTA Check Method: I would add a checkCTA method to handle checks for both headerCTA and footerCTA.

**Review by rgrnearform — COMMENTED** [review]
Added some comments, please check

File: apps/payments/e2e/specs/landingPage/llandingPage.spec.ts
Comment (nnorbert) [reply apps/payments/e2e/specs/landingPage/llandingPage.spec.ts:1]: No, it is a typeo, good catch! Thanks!

Comment (nnorbert) [reply apps/payments/e2e/specs/landingPage/llandingPage.spec.ts:1]: Done!

File: apps/payments/e2e/objects/LandingPage.ts (hunk 1 above)
Comment (nnorbert) [reply apps/payments/e2e/objects/LandingPage.ts:14]: Done!

File: apps/payments/e2e/objects/LandingPage.ts (hunk 2 above)
Comment (nnorbert) [reply apps/payments/e2e/objects/LandingPage.ts]: Done!

**Review by rgrnearform — DISMISSED** [review]
lgtm

**Review by peschina — APPROVED** [review]
LGTM

File: apps/payments/e2e/specs/landingPage/landingPage.spec.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    await tags("Landing page");
+    await severity(Severity.NORMAL);
+
+    const landingPage = new LandingPage(page);
+    await landingPage.goto();
+
+    await testLandingPage(landingPage);
+  });
+
+  test("should test the Payments landing page 2 @smoke @normal", async () => {
```
[the remark this answers is not in the document]
Comment (nnorbert) [reply apps/payments/e2e/specs/landingPage/landingPage.spec.ts]: Done

**Review by peschina — APPROVED** [review]

**Pull request merged**